### ISSUE TITLE
Keyboard input not functioning correctly on iOS Safari

### ISSUE THREAD

**Issue by Legomite** [photonstorm/phaser#4394]
## Version

* Phaser Version: 3.16.2

* Operating system: iOS

* Browser: Mobile Safari

## Description

Keyboard input does not behave as expected on iOS. This was tested using a plugged in keyboard tested on an iPad and iPhone. This can be demonstrated using the Phaser keyboard example on Phaser labs. It appears that the keydown event is continuously fired, and keyup does not occur. I tested the keyboard input on another website: https://unixpapa.com/js/testkey.html to ensure that Safari is sending the correct events and everything seems fine.

## Example Test Code
https://labs.phaser.io/view.html?src=src\input\keyboard\add%20key%20using%20string.js

This demonstrates how the event is not functioning properly. Normally, pressing any keys will cause the letter to flash, and letting go will cause it to revert. On iOS this is retained and does not revert. This behavior occurs throughout all of the keyboard examples.

**Comment by photonstorm** [comment]
This is not something I'm going to fix personally as it's such an unusual edge-case, but I'll happily merge PRs that address it.

**Comment by IxquitilisSaid** [comment]
@Legomite hey dude, I'm checking out this bug and yeah, even on XCode's simulator this bug can be reproduced, but I still need to ask about it though, if you don't mind.

Is this happening just on Phaser Labs or did you run into it during a regular project?

I have my suspicions regarding what it can be but I don't want to point any fingers just yet, but let's say that it might be related to how iOS handles external input events (big maybe)

**Comment by Legomite** [comment]
@IxquitilisSaid I first discovered it while looking to see if I could control a game I was making that was built primarily on a Mac. The keyboard controls functions correctly on the laptop and desktops that I tested. On the iPad I noticed that the game wouldn’t register the key releases, so my player would just continuously walk or jump. I went on Labs to see if it was my fault, but I noticed it was acting weird there as well

**Comment by Legomite** [comment]
Strange... I checked back on my game and the keyboard started to work properly. I suspected it was the new Phaser update, so I went through the keyboard examples and it worked from 3.10 all the way to current. I’m thinking that it’s an issue with iOS. I don’t recall the version of iOS that I was running when I reported this bug, but I’m on 12.2 now and it’s working perfectly fine.

**Comment by photonstorm** [comment]
@Legomite thanks for the update. You're right, maybe it was an iOS change rather than a Phaser one. Either way, I'm glad it's working now! So going to close this issue off.

**Comment by duquesnay** [comment]
I know this bug is closed, but I observe today the same behaviour on Safari on MacOS Monterey 12.3.1 (Safari 15.4). It's exactly as described, the down event is fired forever, even if it's up. I checked the examples and find the same bug (events never stop).
Any clue?
Update: no problem with Chrome. I've checked on other raw JS attempt, I don't see such problem. It looks like the event queue goes crazy
